extensions/wikia/Email/Controller/UserRightsChangedController.class.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\Check;
+use \Email\EmailController;
+
+class UserRightsChangedController extends EmailController {
+
+	private $details;
+
+	public function initEmail() {
+		$this->details = $this->getVal( 'summary' );
+		$this->assertValidParams();
+	}
+
+	private function assertValidParams() {
+		if ( empty( $this->details ) ) {
+			throw new Check( "'summary' parameter must not be empty" );
+		}
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getMessage( "emailext-user-rights-changed-subject" )->parse(),
+			'details' => $this->details,
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'buttonText' => $this->getMessage( 'emailext-user-rights-changed-button-text' )->text(),
+			'buttonLink' => $this->getButtonLink()
+		] );
+	}
+
+	protected function getSubject() {
+		return $this->getMessage( "emailext-user-rights-changed-subject" )->parse();
+	}
+
+
+	private function getButtonLink() {
+		return \SpecialPage::getTitleFor( 'Log' )->getFullURL( [ "type" => "rights", "page" => $this->targetUser->getName() ] );
```

Comment:
This line exceeds our 120 char limit now (https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#line-length).  I'd consider breaking it down at the ```->``` or putting the URL params into a var.